File: HRMS/HRMS.Web/Modules/Recruitment/JobOpenings/JobOpeningsColumns.cs
using Serenity.ComponentModel;

namespace HRMS.Recruitment.Columns;

[ColumnsScript("Recruitment.JobOpenings")]
[BasedOnRow(typeof(JobOpeningsRow), CheckNames = true)]
public class JobOpeningsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int JobId { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public string DepartmentName { get; set; }
    public string HiringManagerName { get; set; }
    public int Status { get; set; }
    public DateTime CreatedDate { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Recruitment/JobOpeningsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { JobOpeningsRow } from "./JobOpeningsRow";
import { JobOpeningStatus } from "./JobOpeningStatus";

export interface JobOpeningsColumns {
    JobId: Column<JobOpeningsRow>;
    Title: Column<JobOpeningsRow>;
    Description: Column<JobOpeningsRow>;
    DepartmentName: Column<JobOpeningsRow>;
    HiringManagerName: Column<JobOpeningsRow>;
    Status: Column<JobOpeningsRow>;
    CreatedDate: Column<JobOpeningsRow>;
}

export class JobOpeningsColumns extends ColumnsBase<JobOpeningsRow> {
    static readonly columnsKey = 'Recruitment.JobOpenings';
    static readonly Fields = fieldsProxy<JobOpeningsColumns>();
}

[JobOpeningStatus]; // referenced types